client/src/annoMatrix/loader.ts

```diff
@@ -320,23 +330,24 @@ export default class AnnoMatrixLoader extends AnnoMatrix {
 Utility functions below
 */
 
-// @ts-expect-error ts-migrate(7006) FIXME: Parameter 'colSchema' implicitly has an 'any' type... Remove this comment to see the full error message
-function _writableCheck(colSchema) {
-  if (!colSchema?.writable) {
+function _writableCheck(colSchema: ColumnSchema) {
+  if (!colSchema || !("writable" in colSchema && colSchema.writable)) {
```

Comment:
Updated if statement as `writable` does not exist on union type ColumnSchema.

Reply:
it should exist as it is a legit flag in the schema.  CC: @tihuan

Reply:
`if (!colSchema?.writable)` yields the following error:

```
 Property 'writable' does not exist on type 'ArraySchema'.   Property 'writable' does not exist on type 'EmbeddingSchema'.
```

I was under the impression that I needed to narrow the type of `colSchema` (with `"writable" in colSchema`) before I could execute the `colSchema.writable` check.

Let me know if I am missing something here or if you have a preferred method of type guarding/narrowing.

Reply:
The issue is that the function name is more general than the function implementation/usage.

Both `_writableCheck` and `_writableCategoryTypeCheck` are _only_ usable on obs columns. I suggest changing their signature and name to reflect that.  In other wrods:

`_writableCheck` becomes `function _writableObsCheck(obsColSchema: AnnotationColumnSchema): void`

Same change to `_writableCategoryTypeCheck`

Reply:
Udpated [_writableObsCheck](https://github.com/chanzuckerberg/cellxgene/blob/mim/annoMatrix/client/src/annoMatrix/loader.ts#L346) and [_writableObsCategoryTypeCheck](https://github.com/chanzuckerberg/cellxgene/blob/mim/annoMatrix/client/src/annoMatrix/loader.ts#L352).